module/Finna/src/Finna/RecordDriver/SolrEad3.php

```diff
@@ -202,35 +202,41 @@ public function getOrigination()
     public function getOriginationExtended()
     {
         $record = $this->getXmlRecord();
-        if (!isset($record->relations->relation)) {
-            return null;
-        }
+        $name = $id = null;
 
-        foreach ($record->relations->relation as $relation) {
-            $attr = $relation->attributes();
-            foreach (['relationtype', 'href', 'arcrole'] as $key) {
-                if (!isset($attr->{$key})) {
+        if (isset($record->relations->relation)) {
```

Comment:
Näitä voisi ehkä samalla siistiä tyyliin `foreach ($record->relations->relation ?? [] as $relation) {`